### DISCUSSION TITLE
Remove only one device from the large Network

### DISCUSSION THREAD

**Discussion by Pritesh-1011** [openthread/openthread#7697] [Q&A] [answered]
I have a large network is created for the open thread devices. In this structure, I have only one device as the leader (FED) and all others are the child devices (SED). Now I want to remove only one child device from the network and the rest of the other devices should remain connected with the leader. Also, the entry of that child device should be removed from the child table list. Are there any standard ways to do that?

**Comment by jwhui** [comment, marked as answer]
Thread does not currently specify a mechanism for removing individual Thread devices from the Thread network.

One option is to have your application layer implement the logic to remove a given device from the Thread network, but that assumes the device you want to remove is running your application.

Once a given child is removed from the network, its associated child table entry on the parent should eventually timeout within the Child Timeout period.

**Comment by ksathara** [comment]
Hi Jonathan, Is there a way to do this from the FTD side? I want to drop a child from the STD to test child supervision. 

Thanks,
Kaushalya

**Reply by jwhui** [reply]
OpenThread does not currently provide an API to evict an individual child entry. But it is probably something you could add for testing.

**Reply by ksathara** [reply]
Thanks Jonathan, Any APIs I can use to drop a child from child table?

**Reply by jwhui** [reply]
OpenThread does not currently provide an API to evict an individual child entry.

**Reply by ksathara** [reply]
what API we can use to reinitiate MLE attach?

**Reply by ksathara** [reply]
I have done it like this. This is based on how CLI 'thread' command is implemented.
```
err = otThreadSetEnabled(openthread_get_default_instance(), false);		// Feature 13
err = otThreadSetEnabled(openthread_get_default_instance(), true);		// Feature 13
```
This seems working. Is this advisable?

**Reply by jwhui** [reply]
Disabling and enabling Thread will cause the device to reattach.

**Reply by ksathara** [reply]
Thanks. Is there a better way to initiate a MLE reattach?

**Reply by jwhui** [reply]
An alternative approach is to cause the device to reboot.
